clang_format.py

```diff
@@ -20,17 +20,61 @@
 import tempfile
 import urllib.request
 from pathlib import Path
+from typing import Final, Literal, Mapping, Optional, Sequence, Tuple, Union
```

Comment:
`Final` and `Literal` are Python 3.8+

Do we particularly want to drop support for Python 3.7 and down?

Ideally this would be done with a snippet that does:

```python
try:
    from typing import Final, Literal
except ImportError:
    from typing_extensions import Final, Literal
```

However, that would require changing the project into an installable Python module that can deliver dependencies. Incidentally, I have a patch that moves the project to use `pyproject.toml`/Poetry to allow `pre-commit` to install the necessary dependencies such as `typing_extensions`.

This _also_ brings into question what the Python engine requirements for the pre-commit hook are and how we test them...